### PR TITLE
Enh md v2

### LINKED ISSUE #103: insert a schema version into the header
This will be very useful in 3 months when you decide to rename something :wink:

### PR DESCRIPTION
**Pull request by chiahaoliu**
closes #103 , #369

### CODE REVIEW

File: xpdacq/beamtimeSetup.py [hunk 1]
```diff
@@ -58,9 +58,8 @@ def _start_beamtime(PI_last, saf_num, experimenters=[],
               "please do\n"
               ">>> xrun.beamtime = bt")
         # copy default Ni24.D to xpdUser/user_analysis
-        #src = os.path.join(os.path.dirname(__file__), 'Ni24.D')
         src = os.path.join(DATA_DIR, 'Ni24.D')
-        dst = os.path.join(glbl_dict['usrAnalysis_dir'], 'Ni.D')
+        dst = os.path.join(glbl_dict['usrAnalysis_dir'], 'Ni24.D')
```
Comment (CJ-Wright) [inline xpdacq/beamtimeSetup.py]: Is this the Ni that @sbanerjee23 was talking about?

File: xpdacq/simulation.py [hunk 2]
```diff
@@ -51,53 +51,58 @@ def __init__(self, frame_acq_time=0.1, acquire=1):
...
         self.cam = SimulatedCam()
         self._staged = False
-        super().__init__(name, read_fields, fs=fs)
+        super().__init__(name, read_fields, reg=reg)
         self.ready = True  # work around a hack in Reader
 
 
 def build_pymongo_backed_broker():
-    """Provide a function level scoped MDS instance talking to
+    '''Provide a function level scoped MDS instance talking to
```
Comment (CJ-Wright) [inline xpdacq/simulation.py]: we may consider just importing this from `databroker.tests.confest`

File: xpdacq/beamtimeSetup.py (hunk 1 above)
Comment (chiahaoliu) [inline xpdacq/beamtimeSetup.py]: yes. though the PR about removing copyright is not there yet. (I might just do it later)

File: xpdacq/simulation.py (hunk 2 above)
Comment (chiahaoliu) [inline xpdacq/simulation.py]: Agree. I was planning to fix everything in ``xpdSim`` where is the place we will import simulation objects eventually.

File: xpdacq/simulation.py [hunk 3]
```diff
@@ -51,53 +51,58 @@ def __init__(self, frame_acq_time=0.1, acquire=1):
 
 
 # define simulated PE1C
-class SimulatedPE1C(be.ReaderWithFileStore):
+class SimulatedPE1C(be.ReaderWithRegistry):
```
Comment (CJ-Wright) [inline xpdacq/simulation.py]: Pull from xpdSim?

File: xpdacq/simulation.py [hunk 4]
```diff
@@ -180,7 +185,8 @@ def insert_imgs(mds, fs, n, shape, save_dir=tempfile.mkdtemp()):
 
 # instantiate simulation objects
 db = build_pymongo_backed_broker()
-db.fs.register_handler('RWFS_NPY', be.ReaderWithFSHandler)
-pe1c = SimulatedPE1C('pe1c', {'pe1_image': lambda: np.ones((5, 5))}, fs=db.fs)
+db.fs.register_handler('RWFS_NPY', be.ReaderWithRegistryHandler)
```
Comment (CJ-Wright) [inline xpdacq/simulation.py]: Should this func use an xrun?

File: xpdacq/simulation.py (hunk 3 above)
Comment (chiahaoliu) [inline xpdacq/simulation.py]: yup, plan to use ``xpdSim`` after PR is pulled in

Comment (CJ-Wright) [inline xpdacq/simulation.py]: Cool

**Pull request merged**